### ISSUE TITLE
setting rst_inname breaks restarts

### ISSUE THREAD

**Issue by andreas-h** [wrf-model/WRF#975]
I'm using WRF v4.1.2.  Trying to restart runs from restart files which are not in the default location doesn't work.

My very first run has

    restart                             = .false.,
    rst_outname                         = '../data/test.runwrf/wrfrst_d<domain>_<date>',

The `wrfrst_` file gets created in the correct location.

Now, when I try to restart, using

    restart                             = .false.,
    rst_inname                         = '../data/test.runwrf/wrfrst_d<domain>_<date>',

the run doesn't even start; the error log looks like this:

       alloc_space_field: domain            1 ,               21028052  bytes allocated
      wrf: calling model_to_grid_config_rec
      wrf: calling set_scalar_indices_from_config
      wrf: calling init_wrfio
      module_io.F: in wrf_ioinit
      Entering ext_gr1_ioinit
       DEBUG wrf_timetoa():  returning with str = [2015-08-11_00:00:00]
       DEBUG wrf_timetoa():  returning with str = [2015-08-11_00:00:00]
       DEBUG wrf_timetoa():  returning with str = [2015-08-12_00:00:00]
       DEBUG wrf_timeinttoa():  returning with str = [0000000000_000:006:000]
     DEBUG setup_timekeeping():  clock after creation,  clock start time = 2015-08-11_00:00:00
     DEBUG setup_timekeeping():  clock after creation,  clock current time = 2015-08-11_00:00:00
     DEBUG setup_timekeeping():  clock after creation,  clock stop time = 2015-08-12_00:00:00
     DEBUG setup_timekeeping():  clock after creation,  clock time step = 0000000000_000:006:000
       setup_timekeeping:  set xtime to   0.0000000E+00
       setup_timekeeping:  set julian to    222.0000
      setup_timekeeping:  returning...
       DEBUG wrf_timetoa():  returning with str = [2015-08-11_00:00:00]
    forrtl: severe (66): output statement overflows record, unit -5, file Internal List-Directed Write
    Image              PC                Routine            Line        Source             
    wrf.exe            00000000031AF86B  for__io_return        Unknown  Unknown
    wrf.exe            00000000031FF3FD  for_write_int_lis     Unknown  Unknown
    wrf.exe            00000000013C2244  Unknown               Unknown  Unknown
    wrf.exe            00000000004155F1  Unknown               Unknown  Unknown
    wrf.exe            00000000004143B5  Unknown               Unknown  Unknown
    wrf.exe            0000000000414352  Unknown               Unknown  Unknown
    libc-2.17.so       00002B3759EC0B15  __libc_start_main     Unknown  Unknown
    wrf.exe            0000000000414269  Unknown               Unknown  Unknown

The exact same setup works correctly when I don't specify the `rst_inname` and `rst_outname` parameters.  But then the `wrfrst_` files get created right in the `run/` directory which I don't want.

I believe this is a bug.  Either I don't understand the meaning of the `rst_inname` parameter -- in that case it should be better documented.  Or there is actually a bug in the code.

Please let me know if you need any more information.

**Comment by weiwangncar** [comment]
Have you tried the following when you make the restart run:

restart                             = .true.,
rst_inname                         =
'../data/test.runwrf/wrfrst_d<domain>_<date>',

**Comment by andreas-h** [comment]
Yes, sorry, that was a typo (here in the Github issue, not in my namelist), which I had already fixed (but Github doesn't send email notifications on editing an issue it seems).

**Comment by catemgn** [comment]
I have a similar problem.
When I try to run with restart, I get a segmentation error like @andreas-h. 
I am using version 4.1.2  My settings are:

restart                             = .true.,
restart_interval                = 10080,
override_restart_timers   = .false., 
write_hist_at_0h_rst         = .false.,

When not doing a restarted simulation, it works perfectly fine. Have you found a bug to fix or is there a way to go around this?

Thanks in advance.
